### DISCUSSION TITLE
Must bind groups start from 0 and be contiguous?

### DISCUSSION THREAD

**Discussion by mobile-bungalow** [gfx-rs/wgpu#4608] [Q&A] [answered]
let's say I have a shader.

```wgsl
@group(1) @binding(0)
var<uniform> ss: ShrimpleStruct;

@group(3) @binding(0)
var<uniform> ppt: PrawnPlicatedType;
```

while rendering I can call 
```rust
rpass.set_bind_group(1, ss_bind_group);
rpass.set_bind_group(3, ppt_bind_group);
```
But I can't construct a pipeline with the same mapping unless I have two empty dummy bind group layouts in between my sets in my pipeline descriptor. Is this an intentional API decision? Could a pipeline not portably support discontinuous groups?

**Comment by 9SMTM6** [comment, marked as answer]
AFAIK the WebGPU API doesnt allow for that. For the pipeline-layout you pass in the bindgroup-layouts as an array, so the index in that array should correspond with the bindgroup-index.

Only the entries in the bindgroup have their indexes explicitly stated, and I believe you can actually skip entries there.

**Reply by 9SMTM6** [reply]
Also consider that the default limit for the number of bindgroups (maxBindGroups) is quite low with 4, while maxBindingsPerBindGroup is pretty much not reachable in a reasonable project (https://www.w3.org/TR/webgpu/#limits).

AFAIK Bindgroups are meant to give some indication of the frequency of change, for optimization, not of belonging together.

So probably best to rather skip in the binding index.
